numpy/f2py/_backends/_meson.py

```diff
@@ -92,13 +92,13 @@ def libraries_substitution(self) -> None:
 
         self.substitutions["lib_declarations"] = "\n".join(
             [
-                f"{lib} = declare_dependency(link_args : ['-l{lib}'])"
+                f"{lib.replace('.','_')} = declare_dependency(link_args : ['-l{lib}'])"
```

Comment:
```suggestion
                # See gh-26634 for an explanation
                f"{lib.replace('.','_')} = declare_dependency(link_args : ['-l{lib}'])"
```

Reply:
@mathomp4 apologies if my previous comment was unclear, I meant adding an inline comment to this PR/Issue.

Reply:
Ahhh. Sorry about that. I thought you wanted an example for the repo's test suite. :)